### ISSUE TITLE
Cannot deploy resources added by other developers, they do not appear in status.

### ISSUE THREAD

**Issue by dopry** [aws-amplify/amplify-cli#6401]
**Describe the bug**

We use a feature branching workflow and each of our developers works in their own sandbox.  When a developer adds a resource (we tested this with cognito triggers, but I believe it to be generally applicable to all added resources) and I checkout their branch and try to push their configuration to my amplify environment, the resource isn't on my resource list to be deployed.

I did a little sleuthing and it noticed that while the resource is listed in the backend-config.json, it is not present in the amplify-meta.json. If I add an entry for the resource to amplify meta, then it appears in my status. 

I would expect that because the resource is in backend-config.json that if it were missing from the amplify-meta.json it would be added and deployed on amplify push. 

running amplify init after checking out the feature branch seems to resolve this.

**Amplify CLI Version**
tested with both 
* 4.36.2 
* 4.41.2

**To Reproduce**
These are specific to our test scenario

1. Developer A Create a repo with amplify init. sandbox A  
2. Developer A amplify add auth and setup a cognito, push it to sandbox A
3. Developer A Commit Code.
4. Developer B Checkout code, amplify init sandbox B, amplify push
5. Developer B Create Feature Branch
6. Developer B amplify auth update, add a PostConfirmationTrigger,  amplify push to sandbox B
7. Developer B commit and push amplify config updates to branch. 
8. Developer A checkout feature branch
9. Developer A amplify push to Sandbox A, notice the curious lack of the PostConfirmationTrigger

**Expected behavior**
That Developer A could successfully deploy the canonical version of the code to their current environment without needing to run amplify init.

**Comment by attilah** [comment]
@dopry look at the [Sandbox environments](https://docs.amplify.aws/cli/teams/sandbox) page where we document exactly the same workflow. As you also noticed, Amplify carries env specific (sandbox specific) information that is not under source control and when working in a team setup, Amplify environment operations should be used in sync with git branch operations. `amplify-meta.json` contains information about the current env according to your own cloud deployment. The step you are missing I think is the `amplify env checkout` in step 4 instead of init. env checkout rebuilds the meta file based on configuration parameters in resource directories and backend.json (which has env agnostic resource information).

If you think we could make if more clear in the documentation somehow, give some suggestions on how could we improve it based on the problem you ran into.

**Comment by dopry** [comment]
I do not want to checkout a different sandbox. I do not have access to any of the other developers sandboxes. I want to push the configuration in the code to my current sandbox.

**Comment by dopry** [comment]
We do this every day with terraform... `git checkout; terraform apply` as our configuration evolves. I don't understand why this workflow has to be so unwieldy with Amplify. Amplify is really promising, but the developer experience can use a lot of love.

**Comment by dopry** [comment]
I followed your instructions to use `amplfy checkout env` to re-checkout my current environment it just tells me the environment can't be recognized. I have to amplify init again to get it into a working state.

**Comment by dopry** [comment]
I've also done some manual edits to dependencies in my backend-config.json file.... When I do, the dependencies in amplify-meta are taking precedence. I feel like amplify-meta should only be the cache of remote state, while backend-config should be authoritative for the desired state. It seems like that is the root of my issue. I expect the code I've checked from out to be authoritative and deployable to my current environment.  I think this is a fairly consistent expectation for any devops practitioners who work with IAC.

**Comment by kaustavghosh06** [comment]
@dopry Yes, right now the behavior is to run `amplify env checkout <env-name>` to update all the metadata files when adding/updating custom stacks/resource or updating backend-config.json. I'll mark this as an enhancement.

**Comment by dopry** [comment]
Go for it. We abandoned amplify and never plan to revisit it. It was a terrible DX.
